src/main/java/build/buildfarm/instance/memory/DelegateCASMap.java

```diff
@@ -0,0 +1,121 @@
+package build.buildfarm.instance.memory;
+
+import build.buildfarm.common.ContentAddressableStorage;
+import build.buildfarm.common.ContentAddressableStorage.Blob;
+import build.buildfarm.common.Digests;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.devtools.remoteexecution.v1test.Digest;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.Parser;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+class DelegateCASMap<K,V extends Message> implements Map<K,V>
```

Comment:
Should this be a ConcurrentMap?

Reply:
IMHO making this a ConcurrentMap is required if either of the following cases applies:
1. Only extending from ConcurrentMap connotes to users that this store is thread-safe
2. Specific features from the ConcurrentMap interface are in use (by users) and likely do not represent trivial method implementations

Regarding 1, Since it is a delegator, I'm inclined to say that it is only as thread safe as the ContentAddressableStorage, which as I'll comment below, is fully expected to be thread-safe, but I could see motivations in either direction.  Adding additional fluff methods just to satisfy that interface is not something that motivates me strongly in the direction of 2